File: MaxProfit.cs
public class Solution {
    public int MaxProfit(int[] prices)
    {
        if (prices.Length == 0) return 0;

        int min = int.MaxValue;
        int max = int.MinValue;

        for(int i = 0; i < prices.Length; i++)
        {
            min = Math.Min(min, prices[i]);
            max = Math.Max(max, prices[i] - min);
        }

        return max;
    }
}

File: Python/maxProfit.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        maxDiff = 0
        left = 0
        
        for right in range(1, len(prices)):
            if prices[right] < prices[left]:
                left = right
            
            maxDiff = max(maxDiff, prices[right] - prices[left])

        return maxDiff
